p2p/src/network/scheduler_effectful/p2p_network_scheduler_effectful_effects.rs

```diff
@@ -0,0 +1,82 @@
+use redux::ActionMeta;
+use std::net::SocketAddr;
+
+use crate::{MioCmd, P2pCryptoService, P2pMioService};
+
+use super::{super::*, *};
+
+impl P2pNetworkSchedulerEffectfulAction {
+    pub fn effects<Store, S>(self, _meta: &ActionMeta, store: &mut Store)
+    where
+        Store: crate::P2pStore<S>,
+        Store::Service: P2pMioService + P2pCryptoService,
+    {
+        match self {
+            P2pNetworkSchedulerEffectfulAction::InterfaceDetected { ip, port } => {
+                store
+                    .service()
+                    .send_mio_cmd(MioCmd::ListenOn(SocketAddr::new(ip, port)));
+            }
+            P2pNetworkSchedulerEffectfulAction::IncomingConnectionIsReady { listener, .. } => {
+                let state = store.state();
+                if state.network.scheduler.connections.len()
+                    >= state.config.limits.max_connections()
+                {
+                    store.service().send_mio_cmd(MioCmd::Refuse(listener));
+                } else {
+                    store.service().send_mio_cmd(MioCmd::Accept(listener));
+                }
+            }
+            P2pNetworkSchedulerEffectfulAction::IncomingDidAccept { addr, .. } => {
+                let nonce = store.service().generate_random_nonce();
```

Comment:
This is for later (out of scope for this PR), but we will need to split this eventually. Any result that is not deterministic will need to "re-enter" the state machine through an event. So eventually we will be able to implement replays with fully mocked services by replaying the events (since all non-deterministic results will be contained there).

Reply:
We have a Crypto service we could use, but we also had some problems with the use of entropy in the p2p layer to keep things working with the network debugger. Maybe @vlad9486 can add some insight.